### PR TITLE
[SYCL] Improve testing of host-task

### PR DESCRIPTION
**Pull request by s-kanaev**
This patch introduces two use-cases for host-task implementation.

The issue of the first use-case (`test3`) was about performance and a long-run/wait of host-taske depening on another host-task. This issue rooted in lots of enqueue calls for blocked command and was fixed in bc8f0a4 (#2347).

The second use-case shows a deadlock situation when host-task `A` depends on the blocked one `B` via call `handler::depends_on()`. The `A` host-task won't be enqueued as it's `EmptyCommand` is in blocked state.
The test for this use-case is disable now and will be enabled when the deadlock is fixed.

The deadlock will be fixed in follow-up patch.

### CODE REVIEW

File: sycl/test/host-interop-task/host-task.cpp [hunk 1]
```diff
@@ -1,34 +1,59 @@
...
+// RUN: %GPU_RUN_PLACEHOLDER %t.out 2
+// RUN: %ACC_RUN_PLACEHOLDER %t.out 2
+
+// RUN: %CPU_RUN_PLACEHOLDER %t.out 3
+// RUN: %GPU_RUN_PLACEHOLDER %t.out 3
+// RUN: %ACC_RUN_PLACEHOLDER %t.out 3
+
+// RUNx: %CPU_RUN_PLACEHOLDER %t.out 4
+// RUNx: %GPU_RUN_PLACEHOLDER %t.out 4
+// RUNx: %ACC_RUN_PLACEHOLDER %t.out 4
```
Comment (s-kanaev) [inline sycl/test/host-interop-task/host-task.cpp:16]: These are going to be enabled after the issue is fixed.

File: sycl/test/host-interop-task/host-task.cpp [hunk 2]
```diff
@@ -55,10 +80,159 @@ void test2() {
...
+        X ^= reinterpret_cast<uint64_t>(&Acc4[Idx + 4]);
+        X ^= reinterpret_cast<uint64_t>(&Acc5[Idx + 5]);
+        X ^= reinterpret_cast<uint64_t>(&Acc6[Idx + 6]);
+        X ^= reinterpret_cast<uint64_t>(&Acc7[Idx + 7]);
+        X ^= reinterpret_cast<uint64_t>(&Acc8[Idx + 8]);
+        X ^= reinterpret_cast<uint64_t>(&Acc9[Idx + 9]);
+
+        std::cout << "  Start " << Idx << " (" << X << ")" << std::endl;
+        using namespace std::chrono_literals;
+        std::this_thread::sleep_for(100ms);
```
Comment (dm-vodopyanov) [inline sycl/test/host-interop-task/host-task.cpp:140]: Why sleep is needed here?

Comment (s-kanaev) [reply sycl/test/host-interop-task/host-task.cpp:140]: It was originally in customer's code. Though, I don't really know if it's needed here.

Comment (s-kanaev) [reply sycl/test/host-interop-task/host-task.cpp]: Removed

File: sycl/test/host-interop-task/host-task.cpp [hunk 3]
```diff
@@ -1,34 +1,59 @@
-// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out %threads_lib -lOpenCL
-// RUN: %CPU_RUN_PLACEHOLDER %t.out
-// RUN: %GPU_RUN_PLACEHOLDER %t.out
-// RUN: %ACC_RUN_PLACEHOLDER %t.out
+// RUN: %clangxx -std=c++14 -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
```
[the remark this answers is not in the document]
Comment (s-kanaev) [reply sycl/test/host-interop-task/host-task.cpp]: Fixed.

**Comment by DenisBakhvalov** [comment]
LGTM.
nit: you could probably say explicitly that you precommit tests for the problem that you will fix later. And briefly describe the problem in the commit message.

**Review by DenisBakhvalov — APPROVED** [review]

**Comment by s-kanaev** [comment]
>  you could probably say explicitly that you precommit tests for the problem that you will fix later. And briefly describe the problem in the commit message.

Done

**Review by dm-vodopyanov — APPROVED** [review]

**Pull request merged**